pilot/pkg/networking/core/v1alpha3/envoyfilter/rc_patch.go

```diff
@@ -389,6 +394,11 @@ func routeMatch(httpRoute *route.Route, rp *model.EnvoyFilterConfigPatchWrapper)
 	return true
 }
 
-func cloneVhostRouteByRouteIndex(virtualHost *route.VirtualHost, routeIndex int) {
```

Comment:
How come this previous code did not work? Seems the issue is the route is shared, but here we clone the route that is modified?

Reply:
Actually, we shared a slice of routes but not a single route itself.

Reply:
If we just share the slice, isn't it more performant to just shallow copy the array and keep cloneVhostRouteByRouteIndex?

Imagine we have 100 elements. We modify 1. 

In this PR, we deep-copy all 100. I think we can just copy the array (cheap) and the 1

Reply:
I agree. The fix now is based on the code before `cloneVhostRouteByRouteIndex` was introduced. We can simply copy a slice in `cloneVhostRoutes` instead of clone each route. 

If needed, I can submit an optimization to keep cloneVhostRouteByRouteIndex, like:
```
func cloneVhostRoutes(routes []*route.Route) []*route.Route {
	out := make([]*route.Route, len(routes))
	copy(out, routes)
	return out
}
```

Reply:
I think you can just use `slices.Clone(routes)` now, but SGTM!

Reply:
ack, and please review https://github.com/istio/istio/pull/44885